File: run_export.py
#!/usr/bin/env python3
"""
Simple runner script for Vibrent Health API Client

This script makes it easy to run the export without dealing with module paths.
"""

import logging
import os
import sys
from pathlib import Path

from dotenv import load_dotenv

# Add src to Python path
src_path = Path(__file__).parent / "src"
sys.path.insert(0, str(src_path))

# Import the required modules
from vibrent_api_client.core.config import ConfigManager, ConfigurationError
from vibrent_api_client.core.exporter import SurveyDataExporter
from vibrent_api_client.utils.helpers import setup_logging


def main():
    """Main entry point"""
    load_dotenv()

    try:
        # Load configuration
        config_manager = ConfigManager()
        
        # Setup logging
        setup_logging()
        logger = logging.getLogger(__name__)

        # Print license information
        print("Vibrent Health API Client - Reference Implementation")
        print("=" * 50)
        print("⚠️  IMPORTANT: This is a reference implementation only")
        print("   - Not intended for production use without testing")
        print("   - No active support or updates provided")
        print("   - No version compatibility guarantees")
        print("   - Report issues to info@vibrenthealth.com")
        print("   - Use at your own risk")
        print("=" * 50)

        # Initialize and run exporter
        environment = os.getenv("VIBRENT_ENVIRONMENT") or config_manager.get("environment.default")
        
        exporter = SurveyDataExporter(config_manager=config_manager, environment=environment)
        exporter.run_export()

    except ConfigurationError as e:
        print(f"Configuration error: {str(e)}")
        print("\nPlease make sure to:")
        print("1. Replace placeholder URLs in config/vibrent_config.yaml with your actual Vibrent Health API URLs")
        print("2. Set VIBRENT_CLIENT_ID and VIBRENT_CLIENT_SECRET environment variables")
        return 1
    except Exception as e:
        logger.error(f"Application failed: {str(e)}")
        return 1

    return 0


if __name__ == "__main__":
    exit(main())

File: java/src/main/java/com/vibrenthealth/apiclient/RunExport.java
package com.vibrenthealth.apiclient;

import com.vibrenthealth.apiclient.core.ConfigManager;
import com.vibrenthealth.apiclient.core.Constants;
import com.vibrenthealth.apiclient.core.SurveyDataExporter;
import com.vibrenthealth.apiclient.utils.Helpers;
import io.github.cdimascio.dotenv.Dotenv;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.util.Arrays;

/**
 * Main entry point for Vibrent Health API Client
 */
public class RunExport {
    private static final Logger logger = LoggerFactory.getLogger(RunExport.class);

    public static void main(String[] args) {
        try {
            // Parse command line arguments
            String configFile = null;
            if (args.length > 0) {
                configFile = args[0];
            }

            // Load configuration
            ConfigManager configManager = new ConfigManager(configFile);

            // Setup logging
            Helpers.setupLogging();
            logger.info("Vibrent Health API Client - Java Reference Implementation");

            // Print license information
            printLicenseInfo();

            // Initialize and run exporter
            String environment = System.getProperty("VIBRENT_ENVIRONMENT");
            if (environment == null) {
                environment = (String) configManager.get(Constants.ConfigKeys.ENVIRONMENT + "." + Constants.ConfigKeys.DEFAULT);
            }

            SurveyDataExporter exporter = new SurveyDataExporter(configManager, environment, null);
            exporter.runExport();

        } catch (ConfigManager.ConfigurationError e) {
            System.err.println("Configuration error: " + e.getMessage());
            System.err.println("\nPlease make sure to:");
            System.err.println("1. Replace placeholder URLs in shared/config/vibrent_config.yaml with your actual Vibrent Health API URLs");
            System.err.println("2. Set VIBRENT_CLIENT_ID and VIBRENT_CLIENT_SECRET environment variables");
            System.err.println("3. You can specify a custom config file: java -jar vibrent-api-client.jar <path_to_config>");
            System.exit(1);
        } catch (Exception e) {
            logger.error("Application failed: {}", e.getMessage());
            System.exit(1);
        }
    }

    /**
     * Print license and warning information
     */
    private static void printLicenseInfo() {
        System.out.println("Vibrent Health API Client - Reference Implementation");
        System.out.println("=".repeat(50));
        System.out.println("⚠️  IMPORTANT: This is a reference implementation only");
        System.out.println("   - Not intended for production use without testing");
        System.out.println("   - No active support or updates provided");
        System.out.println("   - No version compatibility guarantees");
        System.out.println("   - Report issues to info@vibrenthealth.com");
        System.out.println("   - Use at your own risk");
        System.out.println("=".repeat(50));
    }
} 